File: src/service/core/app/objects.py
"""
SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.  # pylint: disable=line-too-long

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

SPDX-License-Identifier: Apache-2.0
"""

import datetime
from typing import Annotated, List

import pydantic

from src.lib.utils import common


AppNamePattern = Annotated[str, pydantic.Field(pattern=common.APP_NAME_VALIDATION_REGEX)]


class ListEntry(pydantic.BaseModel):
    uuid: str
    name: str
    description: str
    created_date: datetime.datetime
    owner: str
    latest_version: int


class ListResponse(pydantic.BaseModel, extra='forbid'):
    apps: List[ListEntry]
    more_entries: bool


class GetVersionEntry(pydantic.BaseModel):
    version: int
    created_by: str
    created_date: datetime.datetime
    status: str


class GetAppResponse(pydantic.BaseModel, extra='forbid'):
    uuid: str
    name: str
    description: str
    created_date: datetime.datetime
    owner: str
    versions: List[GetVersionEntry]


class EditResponse(pydantic.BaseModel, extra='forbid'):
    uuid: str
    version: int
    name: str
    created_by: str
    created_date: datetime.datetime

File: src/runtime/pkg/args/objects.go
/*
SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

SPDX-License-Identifier: Apache-2.0
*/

package args

import (
	"net/url"
	"time"

	"go.corp.nvidia.com/osmo/runtime/pkg/common"
)

type ExecArgs struct {
	Command         string
	Args            common.ArrayFlags
	Checkpoint      common.ArrayFlags
	SocketPath      string
	UnixTimeout     time.Duration
	UserBinPath     string
	HistoryFilePath string
	RunLocation     string

	// Rsync flags
	EnableRsync        bool
	RsyncReadLimit     int
	RsyncWriteLimit    int
	RsyncPathAllowList string

	CliAutoCompleteScriptPath string
}

type CtrlArgs struct {
	Inputs             common.ArrayFlags
	Outputs            common.ArrayFlags
	InputPath          string
	OutputPath         string
	SocketPath         string
	LogSource          string
	WorkflowServiceUrl url.URL
	RefreshTokenUrl    url.URL
	Workflow           string
	Barrier            string
	GroupName          string
	RetryId            string
	RefreshToken       string
	RefreshScheme      string
	TokenHeader        string
	ConfigLoc          string
	UserConfig         string
	ServiceConfig      string
	MetadataFile       string
	FSxLustreConfig    string
	DownloadType       string
	Timeout            time.Duration
	UnixTimeout        time.Duration
	ExecTimeout        time.Duration
	DataTimeout        time.Duration
	LogsPeriod         int
	LogsBufferSize     int
	CacheSize          int
}
